WowCharComparerWebApp/Controllers/User/RegisterController.cs

```diff
@@ -76,24 +88,71 @@ public bool CheckUsername(string accountName, ComparerDatabaseContext db)
 
         public bool CheckEmail(string email, ComparerDatabaseContext db)
         {
-            return email.Contains("@") && email.Contains(".");
+            try
+            {
+                MailAddress mail = new MailAddress(email);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                //TODO write to user that mail format is incorrect or already is in database
+                return false;
+            }
         }
 
-        public void SendVerificationMail(string userEmail)
+        public void SendVerificationMail(string userEmail, Guid activationGuid)
         {
+            var test = HttpContext.Request.Path;
+
             SmtpClient client = new SmtpClient("smtp.gmail.com", 587)
             {
                 DeliveryMethod = SmtpDeliveryMethod.Network,
                 UseDefaultCredentials = false,
-                Credentials = new NetworkCredential(APIConf.WowCharacterComparerEmail, APIConf.WoWCharacterComparerEmailPassword)
+                Credentials = new NetworkCredential(APIConf.WowCharacterComparerEmail, APIConf.WoWCharacterComparerEmailPassword),
+                Timeout = 100000,
+                EnableSsl = true             
             };
 
-            MailMessage msg = new MailMessage();
+            string protocol = HttpContext.Request.IsHttps ? "https" : "http";
+
+            string activationLink = $"{protocol}://www.{ HttpContext.Request.Host}/Register/AccountConfirmation";
+
+            MailMessage msg = new MailMessage();    
             msg.To.Add(userEmail);
             msg.From = new MailAddress(APIConf.WowCharacterComparerEmail);
-            msg.Subject = "Verification";
-            msg.Body = "This is a email to verification your World of Warcraft Character Comparer account";
+            msg.IsBodyHtml = true;
+            msg.Subject = "Verification";          
```

Comment:
Where can be some more info about the subject of the email. "Verification" tells user not much.

Reply:
Added more information to msg.Subject